contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,860 @@
+        upperTick = _upperBoundingTick;
+        tickSpacing = 1;
+    }
+
+    /**
+     * @notice initialize function, to set up initial internal state
+     * @param _rewardTokenAddresses Address of reward token for platform
+     * @param _assets Addresses of initial supported assets
+     * @param _pTokens Platform Token corresponding addresses
+     * @param _clGauge Address of the Aerodrome slipstream pool gauge
+     */
+    function initialize(
+        address[] memory _rewardTokenAddresses,
+        address[] memory _assets,
+        address[] memory _pTokens,
+        address _clGauge
+    ) external onlyGovernor initializer {
+        InitializableAbstractStrategy._initialize(
+            _rewardTokenAddresses,
+            _assets,
+            _pTokens
+        );
+        // TODO: move to constructor once this is known on the mainnet
+        clGauge = ICLGauge(_clGauge);
+    }
+
+    /***************************************
+                  Configuration 
+    ****************************************/
+
+    /**
+     * TODO: delete once we get the gauge.
+     */
+    function setGauge(address _clGauge) external onlyGovernor {
+        clGauge = ICLGauge(_clGauge);
+    }
+
+    /**
+     * @notice Set the new desired WETH share
+     * @param _amount The new amount specified in basis points
+     */
+    function setPoolWethShare(uint256 _amount) external onlyGovernor {
+        // TODO tests:
+        // - governor can update
+        // - non governor can not update
+        // - must be within allowed values (event emitted)
+
+        require(_amount < 1e18, "Invalid poolWethShare amount");
+
+        poolWethShare = _amount;
+        emit PoolWethShareUpdated(_amount);
+    }
+
+    /**
+     * @notice Specifies the amount of liquidity that is to be removed when 
+     *         a rebalancing happens.
+     * @param _amount The new amount specified in basis points
+     */
+    function setWithdrawLiquidityShare(uint128 _amount) external onlyGovernor {
+        // TODO tests:
+        // - governor can update
+        // - non governor can not update
+        // - must be within allowed values (event emitted)
+
+        require(_amount < 1e18, "Invalid withdrawLiquidityShare amount");
+
+        withdrawLiquidityShare = _amount;
+        emit WithdrawLiqiudityShareUpdated(_amount);
+    }
+
+    /**
+     * @notice Specifies how the target WETH share of the pool defined by the `poolWethShare` can
+     *         vary from the configured value after rebalancing.
+     * @param _amount The new amount specified in basis points
+     */
+    function setPoolWethShareVarianceAllowed(uint256 _amount) external onlyGovernor {
+        // TODO tests:
+        // - governor can update
+        // - non governor can not update
+        // - must be within allowed values (event emitted)
+
+        // no sensible reason to ever allow this over 40%
+        require(_amount < 4e17, "Invalid poolWethShareVariance");
```

Comment:
`4e17` confused me for a bit. Can we just use `0.4 ether` here? And probably change all instances of `1e18` to `1 ether`? That would make it much more readable

Reply:
changed it to 0.4 ether. Left the other instances of 1e18 to 1e18, since IMO it is more readable when it comes to shares (1e18 representing 100% share) and writing `1 ether` might confuse the reader that the number represents a token amount instead of percentage: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/9047df5eef6eb2e8de7c0c307b5932a418e7550b

I am open to being persuaded to the contrary